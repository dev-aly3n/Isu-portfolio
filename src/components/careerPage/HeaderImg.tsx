// libs
import { motion } from "framer-motion";
import { RefObject, useRef } from "react";
import { useRouter } from "next/router";
// components
// hooks
// store
import { careerHeaderImage } from "../../store/allData";
// utils & animation
import { fancyBtns } from "../../utils/animation";
import { fancyBtnHandler } from "../../utils/util";
import Image from "next/image";

interface props {
  allRefs: {
    bioRef: RefObject<HTMLDivElement>;
    eduRef: RefObject<HTMLDivElement>;
    skillsRef: RefObject<HTMLDivElement>;
    contactRef: RefObject<HTMLDivElement>;
  };
}

const HeaderImg: React.FC<props> = ({ allRefs }) => {
  const router = useRouter();
  const bioBtnRef = useRef<HTMLButtonElement>(null);
  const eduBtnRef = useRef<HTMLButtonElement>(null);
  const skillsBtnRef = useRef<HTMLButtonElement>(null);
  const contactBtnRef = useRef<HTMLButtonElement>(null);
  const projectBtnRef = useRef<HTMLButtonElement>(null);

  const bioHandler = fancyBtnHandler(bioBtnRef, allRefs.bioRef);
  const skillsHandler = fancyBtnHandler(skillsBtnRef, allRefs.skillsRef);
  const eduHandler = fancyBtnHandler(eduBtnRef, allRefs.eduRef);
  const contactHandler = fancyBtnHandler(contactBtnRef, allRefs.contactRef);
  const projectHandler = () => {
    projectBtnRef.current!.style.left = "120%";
    setTimeout(() => {
      router.push("/projects");
    }, 100);
  };

  return (
    <motion.div layout className="header-img-container">
      <span>
        <img
          alt={careerHeaderImage.alt}
          src={careerHeaderImage.src}
        />
      </span>
      <motion.button
        variants={fancyBtns}
        custom={4}
        onClick={bioHandler}
        ref={bioBtnRef}
      >
        <h3>Biography</h3>
      </motion.button>
      <motion.button
        variants={fancyBtns}
        custom={8}
        onClick={eduHandler}
        ref={eduBtnRef}
      >
        <h3>Education</h3>
      </motion.button>
      <motion.button
        variants={fancyBtns}
        custom={5}
        onClick={skillsHandler}
        ref={skillsBtnRef}
      >
        <h3>Skills</h3>
      </motion.button>
      <motion.button
        variants={fancyBtns}
        custom={5}
        onClick={contactHandler}
        ref={contactBtnRef}
      >
        <h3>Contact</h3>
      </motion.button>
      <motion.button
        variants={fancyBtns}
        custom={8}
        onClick={projectHandler}
        ref={projectBtnRef}
      >
        <h3>Project</h3>
      </motion.button>
    </motion.div>
  );
};

export default HeaderImg;
