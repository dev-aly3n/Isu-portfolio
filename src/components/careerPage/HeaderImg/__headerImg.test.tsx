import { render, screen } from "@testing-library/react";
import HeaderImg from ".";
import React, { useRef } from "react";
import { RefObject } from "react";

const ref = new Object() as RefObject<HTMLDivElement>;
const allRefs = {
  bioRef: ref,
  eduRef: ref,
  skillsRef: ref,
  contactRef: ref,
};

describe("career/Skill", () => {
  it("render heads", () => {
    render(<HeaderImg allRefs={allRefs} />);
    const skillHead = screen.getByText("Education");
    expect(skillHead).toBeInTheDocument();
  });
});
