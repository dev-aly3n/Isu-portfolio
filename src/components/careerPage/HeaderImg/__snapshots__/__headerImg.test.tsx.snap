// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`career/headerImg snap 1`] = `
<div>
  <div
    class="header-img-container"
  >
    <span>
      <img
        alt="dev aly3n javascript developer"
        src="/media/aly3n.jpg"
      />
    </span>
    <button>
      <h3>
        Biography
      </h3>
    </button>
    <button>
      <h3>
        Education
      </h3>
    </button>
    <button>
      <h3>
        Skills
      </h3>
    </button>
    <button>
      <h3>
        Contact
      </h3>
    </button>
    <button>
      <h3>
        Project
      </h3>
    </button>
  </div>
</div>
`;
